server/StudySharp.API/Controllers/CourseController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using StudySharp.API.Requests.Courses;
+using StudySharp.API.Responses.Courses;
+using StudySharp.ApplicationServices.Commands;
+using StudySharp.ApplicationServices.Queries;
+using StudySharp.Domain.General;
+
+namespace StudySharp.API.Controllers
+{
+    [Authorize]
+    [ApiController]
+    [Route("api/[courses]")]
+
+    public class CourseController : ControllerBase
```

Comment:
Probably we should add GetCoursesByUserIdQuery (because it was present in the CourseRepository), a handler for it, and a method to the controller. Also, corresponding Request/Response models